src/Types/Scripts.purs

```diff
@@ -32,6 +36,14 @@ derive newtype instance Ord PlutusScript
 instance Show PlutusScript where
   show = genericShow
 
+-- This instance is needed as the server will return a hex-encoded CBOR string
+-- when `applyArgs` is called
+instance DecodeJson PlutusScript where
+  decodeJson j = Json.caseJsonString
+    (Left (Json.TypeMismatch "expected a hex-encoded CBOR string"))
+    (rmap wrap <<< note (Json.UnexpectedValue j) <<< hexToByteArray)
```

Comment:
Could `map` instead of `rmap`,
or
 `(maybe (Left $ Json.UnexpectedValue j) (Right <<< wrap) <<< hexToByteArray)`